### PR TITLE
Port features and improvements from Spacefarers

### PR DESCRIPTION
**Pull request by out-of-phaze**
- Improves the function of the 'cycle to external air' functionality on airlocks.
- Adds the force_open argument to various NanoUI update procs.
- Refactors overmap contact handling somewhat.
- Adds the ability to suppress the 'not available at roundstart' warning in the job preferences menu, for e.g. modes/downstreams where there are no roundstart roles to begin with.
- Adds nutriment sheets, useful for stocking a food replicator given that you can't recycle food into it.
- Allows putting hats on top of helmets.
image

### CODE REVIEW

File: code/modules/clothing/head/misc.dm [hunk 1]
```diff
@@ -46,13 +48,15 @@
...
 	siemens_coefficient = 0.9
+	accessory_slot = null // cannot be equipped on top of helmets
 
 /obj/item/clothing/head/hasturhood
 	name = "hastur's hood"
 	desc = "It's unspeakably stylish."
 	icon = 'icons/clothing/head/hastur.dmi'
 	flags_inv = BLOCK_HEAD_HAIR
 	body_parts_covered = SLOT_HEAD|SLOT_FACE|SLOT_EYES
+	accessory_slot = null // cannot be equipped on top of helmets
```
Comment (MistakeNot4892) [inline code/modules/clothing/head/misc.dm:59]: A lot of these seem a bit arbitrary to restrict, like that ridiculous wig. If you can equip a hairflower why can't you equip a hood?

File: code/modules/materials/material_sheets_mapping.dm [hunk 2]
```diff
@@ -96,6 +96,7 @@ STACK_SUBTYPES(blackbronze,    "black bronze",                  solid/metal/blac
 STACK_SUBTYPES(redgold,        "red gold",                      solid/metal/redgold,         ingot,            null)
 STACK_SUBTYPES(stainlesssteel, "stainless steel",               solid/metal/stainlesssteel,  ingot,            null)
 STACK_SUBTYPES(ice,            "ice",                           liquid/water,                cubes,            null)
+STACK_SUBTYPES(nutriment,      "nutriment",                     liquid/nutriment,            segment,          null)
```
Comment (MistakeNot4892) [inline code/modules/materials/material_sheets_mapping.dm:99]: Nutriment sheets??

Comment (MistakeNot4892) [reply code/modules/materials/material_sheets_mapping.dm:99]: Ah, for replicator.

File: code/modules/clothing/head/misc.dm (hunk 1 above)
Comment (out-of-phaze) [reply code/modules/clothing/head/misc.dm:59]: i think the concern was sprites, the hood might look weird on top of a wider helmet. it's purely visual

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**